packages/cli/src/commands/components/pull/actions.ts

```diff
@@ -9,11 +9,14 @@ export const fetchComponents = async (space: string): Promise<SpaceComponent[] |
   try {
     const client = mapiClient();
 
-    const { data } = await client.get<{
-      components: SpaceComponent[];
-    }>(`spaces/${space}/components`, {
+    const { data } = await client.components.list({
+      path: {
+        space_id: Number.parseInt(space, 10),
+      },
+      throwOnError: true,
     });
-    return data.components;
+
+    return data?.components as SpaceComponent[];
```

Comment:
suggestion: I think we should strive to not use `as` as much as possible. In this case we can use a typeguard instead: `if (!data.components) throw new Error('Error message that makes sense')`